Fix:  make logging have a per profile persistent control method

SlySven opened the pull request:
As an long outstanding issue whether a profile was logged or not on startup was down to a centralised mechanism that worked on every profile loaded - when it should have been a per profile one...

This will close #5872 and close #5114.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

#### Release post highlight
Fix: control over a profile's logging to text/HTML from one session to the next has been improved so that each profile is independent of any other; previously there was a common/shared setting that was saved between Mudlet application runs that meant the last profile used setting was applied to any profile started in the next run.

SlySven committed:
Fix:  make logging have a per profile persistent control method

As an long outstanding issue whether a profile was logged or not on
startup was down to a centralised mechanism that worked on every profile
loaded - when it should have been a per profile one...

This will close #5872.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented:
One, minor niggle, if the profile is closed down whilst logging is active is that although a dividing line is correctly inserted when the logging restarted on a profile load, there is no message before the line indicating when the profile was closed down in the previous session:
![image](https://user-images.githubusercontent.com/6163092/149714659-45c98874-1339-418b-b3bd-0f9a319c3ca2.png)
it would also be nice if the messages could report that the logging was "suspended"/"resumed" rather than "stopped"/"started" when it is being done by the profile being ended/started rather than the logging button / Lua API functions being used...

vadi2 commented on src/XMLexport.cpp:
```diff
@@ -401,7 +401,9 @@ void XMLexport::writeHost(Host* pHost, pugi::xml_node mudletPackage)
     host.append_attribute("mEnableMSDP") = pHost->mEnableMSDP ? "yes" : "no";
     host.append_attribute("mMapStrongHighlight") = pHost->mMapStrongHighlight ? "yes" : "no";
-    host.append_attribute("mLogStatus") = pHost->mLogStatus ? "yes" : "no";
+    // It does not make sense to store this detail in the game save as it now
+    // handled outside of the profile game saves:
+    // host.append_attribute("mLogStatus") = pHost->mLogStatus ? "yes" : "no";
```
I'd just remove it, recall that we're using Git to keep the history instead of the source files.

vadi2 commented on src/XMLimport.cpp:
```diff
@@ -815,7 +815,9 @@ void XMLimport::readHostPackage(Host* pHost)
     pHost->mEnableMSP = attributes().value(qsl("mEnableMSP")) == YES;
     pHost->mMapStrongHighlight = attributes().value(qsl("mMapStrongHighlight")) == YES;
-    pHost->mLogStatus = attributes().value(qsl("mLogStatus")) == YES;
+    // It does not make sense to store this detail in the game save as it now
+    // handled outside of the profile game saves:
+    // pHost->mLogStatus = attributes().value(qsl("mLogStatus")) == YES;
```
Same here.

vadi2 reviewed (approved):
Thank you for jumping on to fix this quick!

vadi2 commented:
Let's only address https://github.com/Mudlet/Mudlet/pull/5874#issuecomment-1014164383 if players see it being as a problem. I don't see any issues with this for example, and the extra code to do this wouldn't really be worth the maintenance load.

SlySven committed:
Revise: remove code commented out in last commit

It is not helpful to store the text/HTML logging enabled/disabled state
in the profile save XML file when it is now properly managed by a sentinel
file in the profile's own part of the file-system.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>
